core/src/main/java/com/radixdlt/messaging/MessagingModule.java

```diff
@@ -237,39 +262,54 @@ RxRemoteEnvironment rxRemoteEnvironment(
       MessageCentralPeerDiscovery messageCentralPeerDiscovery) {
     return new RxRemoteEnvironment() {
       @Override
-      public <T> Flowable<RemoteEvent<T>> remoteEvents(Class<T> remoteEventClass) {
-        if (remoteEventClass == Vote.class) {
-          return messageCentralBFT.remoteVotes().map(m -> (RemoteEvent<T>) m);
-        } else if (remoteEventClass == Proposal.class) {
-          return messageCentralBFT.remoteProposals().map(m -> (RemoteEvent<T>) m);
-        } else if (remoteEventClass == GetVerticesRequest.class) {
-          return messageCentralBFTSync.requests().map(m -> (RemoteEvent<T>) m);
-        } else if (remoteEventClass == GetVerticesResponse.class) {
-          return messageCentralBFTSync.responses().map(m -> (RemoteEvent<T>) m);
-        } else if (remoteEventClass == GetVerticesErrorResponse.class) {
-          return messageCentralBFTSync.errorResponses().map(m -> (RemoteEvent<T>) m);
-        } else if (remoteEventClass == MempoolAdd.class) {
-          return messageCentralMempool.mempoolComands().map(m -> (RemoteEvent<T>) m);
-        } else if (remoteEventClass == SyncRequest.class) {
-          return messageCentralLedgerSync.syncRequests().map(m -> (RemoteEvent<T>) m);
-        } else if (remoteEventClass == SyncResponse.class) {
-          return messageCentralLedgerSync.syncResponses().map(m -> (RemoteEvent<T>) m);
-        } else if (remoteEventClass == StatusRequest.class) {
-          return messageCentralLedgerSync.statusRequests().map(m -> (RemoteEvent<T>) m);
-        } else if (remoteEventClass == StatusResponse.class) {
-          return messageCentralLedgerSync.statusResponses().map(m -> (RemoteEvent<T>) m);
-        } else if (remoteEventClass == LedgerStatusUpdate.class) {
-          return messageCentralLedgerSync.ledgerStatusUpdates().map(m -> (RemoteEvent<T>) m);
-        } else if (remoteEventClass == Ping.class) {
-          return messageCentralPeerLiveness.pings().map(m -> (RemoteEvent<T>) m);
-        } else if (remoteEventClass == Pong.class) {
-          return messageCentralPeerLiveness.pongs().map(m -> (RemoteEvent<T>) m);
-        } else if (remoteEventClass == GetPeers.class) {
-          return messageCentralPeerDiscovery.getPeersEvents().map(m -> (RemoteEvent<T>) m);
-        } else if (remoteEventClass == PeersResponse.class) {
-          return messageCentralPeerDiscovery.peersResponses().map(m -> (RemoteEvent<T>) m);
+      public <N, T> Flowable<RemoteEvent<N, T>> remoteEvents(
```

Comment:
I'm concerned about this if statement. I may be misunderstanding when this is called, but I think this means that every message we receive, we are allocating 20 MessageTransportType classes?

I imagine we'd want to neaten this up?